terraform/dotnet/ec2/default/main.tf

```
@@ -0,0 +1,319 @@
+      sudo dnf install -y dotnet-sdk-8.0
+      sudo yum install unzip -y
+
+      # Copy in CW Agent configuration
+      agent_config='${replace(replace(file("./amazon-cloudwatch-agent.json"), "/\\s+/", ""), "$REGION", var.aws_region)}'
+      echo $agent_config > amazon-cloudwatch-agent.json
+
+      # Get and run CW agent rpm
+      ${var.get_cw_agent_rpm_command}
+      sudo rpm -U ./cw-agent.rpm
+      sudo /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a fetch-config -m ec2 -s -c file:./amazon-cloudwatch-agent.json
+
+      # Get ADOT distro and unzip it
+      ${var.get_adot_distro_command}
+
+      # Get and run the sample application with configuration
+      aws s3 cp ${var.sample_app_zip} ./dotnet-sample-app.zip
+      unzip -o dotnet-sample-app.zip
+
+      # Get Absolute Path
+      current_dir=$(pwd)
+      echo $current_dir
+
+      # Export environment variables for instrumentation
+      cd ./asp_frontend_service
+      export CORECLR_ENABLE_PROFILING=1
+      export CORECLR_PROFILER={918728DD-259F-4A6A-AC2B-B85E1B658318}
+      export CORECLR_PROFILER_PATH=$current_dir/dotnet-distro/linux-x64/OpenTelemetry.AutoInstrumentation.Native.so
+      export DOTNET_ADDITIONAL_DEPS=$current_dir/dotnet-distro/AdditionalDeps
+      export DOTNET_SHARED_STORE=$current_dir/dotnet-distro/store
+      export DOTNET_STARTUP_HOOKS=$current_dir/dotnet-distro/net/OpenTelemetry.AutoInstrumentation.StartupHook.dll
+      export OTEL_DOTNET_AUTO_HOME=$current_dir/dotnet-distro
+      export OTEL_DOTNET_AUTO_PLUGINS="AWS.Distro.OpenTelemetry.AutoInstrumentation.Plugin, AWS.Distro.OpenTelemetry.AutoInstrumentation"
+      export OTEL_EXPORTER_OTLP_PROTOCOL=http/protobuf
+      export OTEL_EXPORTER_OTLP_ENDPOINT=http://127.0.0.1:4316
+      export OTEL_AWS_APPLICATION_SIGNALS_EXPORTER_ENDPOINT=http://127.0.0.1:4316/v1/metrics
+      export OTEL_METRICS_EXPORTER=none
+      export OTEL_RESOURCE_ATTRIBUTES=service.name=dotnet-sample-application-${var.test_id}
+      export OTEL_AWS_APPLICATION_SIGNALS_ENABLED=true
+      export OTEL_TRACES_SAMPLER=always_on
+      export ASPNETCORE_URLS=http://0.0.0.0:8080
+      dotnet build
+      nohup dotnet bin/Debug/netcoreapp8.0/asp_frontend_service.dll &
+
+      # The application needs time to come up and reach a steady state, this should not take longer than 30 seconds
+      sleep 30
+
+      # Check if the application is up. If it is not up, then exit 1.
+      attempt_counter=0
+      max_attempts=30
+      until $(curl --output /dev/null --silent --fail $(echo "http://localhost:8080" | tr -d '"')); do
+        if [ $attempt_counter -eq $max_attempts ];then
+          echo "Failed to connect to endpoint. Will attempt to redeploy sample app."
+          deployment_failed=1
+          break
+        fi
+        echo "Attempting to connect to the remote endpoint. Tried $attempt_counter out of $max_attempts"
```

This is the main service instance
```suggestion
        echo "Attempting to connect to the main endpoint. Tried $attempt_counter out of $max_attempts"
```